includes/in-person-payments/class-wc-payments-in-person-payments-receipts-service.php

```diff
@@ -15,26 +15,16 @@ class WC_Payments_In_Person_Payments_Receipts_Service {
 	/**
 	 * Renders the receipt template.
 	 *
-	 * @param  array    $settings Merchant settings.
-	 * @param  WC_Order $order Order instance.
-	 * @param  array    $charge Charge data.
+	 * @param  array $settings Merchant settings.
+	 * @param  array $order_data Order instance.
+	 * @param  array $charge Charge data.
 	 *
 	 * @return string
 	 */
-	public function get_receipt_markup( array $settings, WC_Order $order, array $charge ) :string {
+	public function get_receipt_markup( array $settings, array $order_data, array $charge ) :string {
```

Comment:
I thought it was simpler to pass array. This way I don't have to create a `WC_Order` instance when I call this one from the preview endpoint. And it also lets me pass only the data that is going to be used to generate the receipt, since the WC_Order has more than it is needed.

Anyway I am going to give it a try and see how it looks. Thanks!

Reply:
@dpaun1985 I tried to follow your suggestion but found some unexpected issues along the way.  The thing is that if I create a fake order to be used by the preview functionality, this order is persisted in the db. I thought I could create a 'memory only' order for this purpose, but it does not seem to be possible.
I tried to create a sample order class for this purpose, extending from WC_Order, but I found I would also need to create a sample item class to be used in this sample order. So it is growing complexity.
I wonder if you are aware of a different approach or if you know a way to achieve this? Thanks in advance

Reply:
@dpaun1985 I tried a different approach here, let me know your thoughts when you have a moment.

Reply:
indeed, we can go with array, as we can't avoid persisting the data in db if we use `WC_Order`